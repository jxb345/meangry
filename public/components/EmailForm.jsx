import React, { useEffect } from "react";

const EmailForm = (props) => {
  const handleSubjectChange = (e) => {
    e.preventDefault();
    props.setSubject(e.target.value);
  };

  const handleBodyChange = (e) => {
    e.preventDefault();
    props.setBody(e.target.value);
  };

  useEffect(() => {
    document.getElementById("subject").value = props.subject;
    document.getElementById("body").value = props.body;
  });

  return (
    <div>
      <input
        type="text"
        id="subject"
        className="subject"
        placeholder="I feel _______"
        onChange={handleSubjectChange}
      />
      <br />
      <textarea
        type="text"
        id="body"
        className="body"
        rows="11"
        cols="69"
        placeholder="because..."
        onChange={handleBodyChange}
      />
      <br />
    </div>
  );
};

export default EmailForm;
