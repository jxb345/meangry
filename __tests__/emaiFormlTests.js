import React from 'react';
import renderer from 'react-test-renderer';
import EmailForm from '../public/components/EmailForm.jsx';

test('emailForm-snapshot-test', () => {
  const component = renderer.create(
    <EmailForm/>);

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    console.log('tree', tree);
})
