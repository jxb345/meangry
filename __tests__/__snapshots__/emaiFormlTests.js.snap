// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`emailForm-snapshot-test 1`] = `
<div>
  <input
    className="subject"
    id="subject"
    onChange={[Function]}
    placeholder="I feel _______"
    type="text"
  />
  <br />
  <textarea
    className="body"
    cols="69"
    id="body"
    onChange={[Function]}
    placeholder="because..."
    rows="11"
    type="text"
  />
  <br />
</div>
`;
